File: calculator.go
package main

import (
	"errors"
	"fmt"
)

type Number float64

func Add(a, b Number) Number {
	return a + b
}

func Subtract(a, b Number) Number {
	return a - b
}

func Multiply(a, b Number) Number {
	return a * b
}

func Divide(a, b Number) (Number, error) {
	if b == 0 {
		return 0, errors.New("除数不能为0")
	}
	return a / b, nil
}

func main() {
	x, y := Number(10), Number(3)
	fmt.Printf("%.0f + %.0f = %.0f\n", x, y, Add(x, y))
	fmt.Printf("%.0f - %.0f = %.0f\n", x, y, Subtract(x, y))
	fmt.Printf("%.0f x %.0f = %.0f\n", x, y, Multiply(x, y))

	result, err := Divide(x, y)
	if err != nil {
		fmt.Println("除零错误:", err)
	} else {
		fmt.Printf("%.0f / %.0f = %.2f\n", x, y, result)
	}
}

File: calculator.py
#!/usr/bin/env python3
"""简单的计算器模块 - 由 Hermes + Claude Code 协同完成"""

from typing import Union

Number = Union[int, float]


def add(a: Number, b: Number) -> Number:
    """加法"""
    return a + b


def subtract(a: Number, b: Number) -> Number:
    """减法"""
    return a - b


def multiply(a: Number, b: Number) -> Number:
    """乘法"""
    return a * b


def divide(a: Number, b: Number) -> Number:
    """除法（含除零保护）"""
    if b == 0:
        raise ValueError("除数不能为0")
    return a / b


def main():
    """演示"""
    x, y = 10, 3
    print(f"{x} + {y} = {add(x, y)}")
    print(f"{x} - {y} = {subtract(x, y)}")
    print(f"{x} x {y} = {multiply(x, y)}")
    print(f"{x} / {y} = {divide(x, y):.2f}")


if __name__ == "__main__":
    main()
